### PR TITLE
`Development`: Move node memory leak workaround

### PR DESCRIPTION
**Pull request by 4ludwig4**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
It seems there have been already multiple workarounds for memory leaks like [in this PR](https://github.com/ls1intum/Artemis/pull/4094).
These workarounds are so far just used for the GitHub Actions (build and some tests) and explained in the docs. 
Therefore, newcomers to the project would have to set `--max-old-space-size` themselves and for each other new build or test via GitHub actions or other build agents you would have to set an environment variable as it has been so far.

### Description
The solution would be to either fix all of the builds by introducing a `.npmrc` with the workaround parameter, or by appending it to each build command for node in `package.json`.
Also, the manual for this parameter has been added and the style of the parameter was matched to the official docs.

### Steps for Testing

- Build the client or both locally and verify GitHub actions are still building as supposed to

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: docs/dev/setup.rst [hunk 1]
```diff
@@ -522,7 +522,8 @@ above in Server Setup) and if you have configured
 with your TUM Online account.
 
 In case you encounter any problems regarding JavaScript heap memory leaks when executing ``npm run serve`` or any other scripts from ``package.json``,
-you can add a memory limit parameter (``--max_old_space_size=5120``) in the script.
+you can add a `memory limit parameter <https://nodejs.org/docs/latest-v16.x/api/cli.html#--max-old-space-sizesize-in-megabytes>`__
```
Comment (ge65cer) [inline docs/dev/setup.rst:525]: Wouldn't that be set by the `.npmrc` config file? Maybe it would be better to direct the reader there.

**Review by ge65cer — COMMENTED** [review]
Generally seems reasonable to adjust this. I have one question:

**Review by akesfeden — COMMENTED** [review]
The configuration changes seem reasonable but it seems the client tests are failing at https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS3774-TSTEST-5. Is this related?

**Comment by dangcosta** [comment]
The workaround worked for me when tested locally. I re-run the tests to see what was the problem. If they are fixed, I will approve the PR

**Comment by 4ludwig4** [comment]
> The configuration changes seem reasonable but it seems the client tests are failing at https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS3774-TSTEST-5. Is this related?

Mhhh first thought this is a problem from somewhere else but it keeps on failing ... Something is not good for the bamboo agent or somehow it's not running now. Anyone has some ideas how I can debug this?

**Comment by 4ludwig4** [comment]
I guess the `client-tests` have some kind of memory problem if you run them with the bigger `--max-old-space-size` necessary for the build and other tests.

Fixed the github `client-tests` by removing the `.npmrc` and therefore not using the now project-wide settings for node for this test.

There are still problems in the `bamboo client-tests` maybe they result from the same issue as the other `client-tests`. I don't know yet how I can change the bamboo tests, anyone knows how to do that?

**Comment by 4ludwig4** [comment]
> There are still problems in the `bamboo client-tests` maybe they result from the same issue as the other `client-tests`. I don't know yet how I can change the bamboo tests, anyone knows how to do that?

Tested the `bamboo client-tests` in a duplicated bamboo job. Removing the `.npmrc` and therefore the node-options also fixes the tests: https://bamboo.ase.in.tum.de/browse/ARTEMIS-ATL0-TSTEST-2/log

Not sure if it would be a cleaner solution to fix theses tests not by removing the `.npmrc` but maybe just removing the node-options line with `sed` or `awk`? Any thoughts on that?

**Comment by sleiss** [comment]
> Not sure if it would be a cleaner solution to fix theses tests not by removing the `.npmrc` but maybe just removing the node-options line with `sed` or `awk`? Any thoughts on that?

Would you remove the node-options with e.g. `sed` within the Bamboo build plan? I would prefer another approach as it is not clearly visible what is happening if it is done within Bamboo.

**Review by JohannesStoehr — DISMISSED** [review]
Tested locally and client successfully builded without any new warnings

**Review by tobias-lippert — APPROVED** [review]
Client built successfully locally. Great that modifications to the package.json to fix the issue are no longer necessary.

**Review by JohannesStoehr — COMMENTED** [review]
The server tests currently fail on GitHub. Can you make sure that this is only a flaky test case and unrelated to this PR?

**Comment by 4ludwig4** [comment]
> The server tests currently fail on GitHub. Can you make sure that this is only a flaky test case and unrelated to this PR?

@JohannesStoehr yeah it seems like the one server test which failed is flaky.

**Review by JohannesStoehr — APPROVED** [review]
Changes look good after merges

**Review by tobias-lippert — APPROVED** [review]
Reapprove

**Review by jpbernius — APPROVED** [review]

**Pull request merged**